crates/energy_optimization/src/current_minimizer.rs

```diff
@@ -0,0 +1,96 @@
+use std::f32::EPSILON;
+
+use serde::{Deserialize, Serialize};
+use serialize_hierarchy::SerializeHierarchy;
+use types::{
+    joints::{body::BodyJoints, head::HeadJoints, Joints},
+    motor_commands::MotorCommands,
+    parameters::CurrentMinimizerParameters,
+};
+
+#[derive(Clone, Copy, Debug, Default, Deserialize, Serialize, SerializeHierarchy)]
+pub enum State {
+    #[default]
+    Optimizing,
+    Resetting,
+}
+
+#[derive(Clone, Copy, Debug, Default, Deserialize, Serialize, SerializeHierarchy)]
+pub struct CurrentMinimizer {
+    pub position_offset: Joints<f32>,
+    pub state: State,
+    pub last_motor_commands: MotorCommands<Joints<f32>>,
+    pub last_positions: Joints<f32>,
+    pub parameters: CurrentMinimizerParameters,
+}
+
+impl CurrentMinimizer {
+    pub fn optimize_body(
```

Comment:
I wasn't sure how to make it more generic, since the `optimization_sign' is loaded from the config and is over `joints<f32>'. You could always load this and discard the parts that aren't needed for the current selection (e.g. an arm).
However, since `optimize_body()` is now removed, I think it's fine as is, and if this functionality is ever wanted, it can be implemented.